[#2181] Merge internalFilename and originalFilename

muyangye opened the pull request:
<!--
  ~ Licensed to the Apache Software Foundation (ASF) under one or more
  ~ contributor license agreements.  See the NOTICE file distributed with
  ~ this work for additional information regarding copyright ownership.
  ~ The ASF licenses this file to You under the Apache License, Version 2.0
  ~ (the "License"); you may not use this file except in compliance with
  ~ the License.  You may obtain a copy of the License at
  ~
  ~    http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~ Unless required by applicable law or agreed to in writing, software
  ~ distributed under the License is distributed on an "AS IS" BASIS,
  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  ~ See the License for the specific language governing permissions and
  ~ limitations under the License.
  ~
  -->
  
  <!--
Thanks for contributing! Here are some tips you can follow to help us incorporate your contribution quickly and easily:
1. If this is your first time, please read our contributor guidelines:
    - https://streampipes.apache.org/getinvolved.html
    - https://cwiki.apache.org/confluence/display/STREAMPIPES/Getting+Started
2. Make sure the PR title is formatted like: `[#<GitHub issue id>] PR title ...`
3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., `[WIP][#<GitHub issue id>] PR title ...`.
4. Please write your PR title to summarize what this PR proposes/fixes.
5. Link the PR to the corresponding GitHub issue (if present) in the `Development` section in the right menu bar. 
6. Be sure to keep the PR description updated to reflect all changes.
7. If possible, provide a concise example to reproduce the issue for a faster review.
8. Make sure tests pass via `mvn clean install`.
9. (Optional) If the contribution is large, please file an Apache ICLA
    - http://apache.org/licenses/icla.pdf
-->

### Purpose
See #2181, finally merging internalFilename and originalFilename to one.
<!--
Please clarify what changes you are proposing and describe how those changes will address the issue.
Furthermore, describe potential consequences the changes might have.
-->

### Remarks
Result:
I created 3 files with the same name using StreamPipes 0.93.0 release:
![14ed46d5129d58af83ba1c22e0a0d52](https://github.com/apache/streampipes/assets/76769570/eeb02a0a-72d0-4cb5-88a0-b1cdc51edc2c)
After which I run the latest migration which is successful:
![4ebe269d0724a23570fa49df7d94a90](https://github.com/apache/streampipes/assets/76769570/377dbf43-da12-41ce-a9c4-ae59d6681b15)
However, for **pipelines, specifically Data Lake (not adapters, adapters can produce messages as usual (i.e. if I create another pipeline using the same adapter created in v0.93.0, I can still visualize it correctly))** created before using File Stream, DataLake expects a new schema:
![508237bb823499217ffdc5b9a4285cc](https://github.com/apache/streampipes/assets/76769570/ccef7b0c-ae63-4095-a8d8-0ba411eca6bc)
After which all pipelines are back to normal:
![84b5655d8abfbdaa62050342c26de28](https://github.com/apache/streampipes/assets/76769570/b2137e4c-e666-414d-8abb-09f0d9876273)


Method:
Get raw json stored in CouchDB => use file ID as the "link" between old version of `FileMetadata` and new version of `FileMetadata` => then update `FileMetadata` and File stored locally.

Test:
I haven't written any unit tests yet because I don't know how to test it since `FileMetadata` class was changed. My original thought is this: create a new class called `oldFileMetadata` which is the same as the `FileMetadata` class that has both originalFilename and internalFilename => store that object using a `CouchDbClient` in the FileMetadata database => run the migration. However, I don't understand why but `Utils.getCouchDbFileMetadataClient()`throws an exception of "unknown host" in file I created `streampipes-service-core/src/test/java/org.apache.streampipes.service.core.migrations.v095/MergeFilenamesAndRenameDuplicatesTest.java` so I deleted it. I believe it's a package/config issue, do you have any idea why? Happy to write a unit test after fixing this issue and everyone agrees with my approach :)

<!--
Is there anything left we need to pay attention on?
Are there some references that might be important? E.g. links to Confluence, or discussions
on the mailing list or GitHub.
-->
PR introduces (a) breaking change(s): yes? Please let me know if the required manual schema update of Data Lake is a problem.

PR introduces (a) deprecation(s): no

tenthe commented:
@muyangye thank you for updating the PR.
I think your approach looks good. 
I just had a couple of small comments.

Regarding the exception in the test, I guess you have to mock the static method somehow, otherwise it tries to connect to the couchdb.

muyangye commented:
@tenthe I will try to see if I can mock the static method somehow. Thanks for pointing out!

Did you leave some small comments on code? For some reason I did not see them?

tenthe commented:
> @tenthe I will try to see if I can mock the static method somehow. Thanks for pointing out!
> 
> Did you leave some small comments on code? For some reason I did not see them?

Sorry, they were still on pending. Can you now see them?

muyangye commented on streampipes-model/src/main/java/org/apache/streampipes/model/file/FileMetadata.java:
```diff
@@ -28,15 +28,16 @@ public class FileMetadata {
 
   private String createdByUser;
 
+  public String test;
```
Yes, this is solely for testing purposes to see what will happen if I add another attribute to the FileMetadata class, will delete.

muyangye commented on streampipes-pipeline-management/src/main/java/org/apache/streampipes/manager/file/FileHandler.java:
```diff
@@ -39,6 +40,16 @@ public File getFile(String filename) {
+      deleteFile(oldFilename);
+      storeFile(newFilename, fileInputStream);
+    } catch (Exception e) {
+      System.out.println(e.getMessage());
```
Both are definitely doable, I will push my changes together with the unit tests.

muyangye commented:
@tenthe I have pushed an update that addressed your code review and added unit tests. I admit that the code is really ugly given that I added an `isTesting` field in the migration and Java doesn't support default parameters. Do you have any idea that can decouple the migration and its test other than recreating all the methods?

muyangye commented:
Hi @tenthe, did you have a chance to review my unit tests?

tenthe reviewed (approved):
@muyangye sorry for the late review. 
Your changes look good. Thanks a lot

bossenti commented on streampipes-service-core/src/main/java/org/apache/streampipes/service/core/migrations/v095/MergeFilenamesAndRenameDuplicatesMigration.java:
```diff
@@ -0,0 +1,217 @@
+    try {
+      return mapper.readValue(inputStream, Map.class);
+    } catch (Exception e) {
+      return new HashMap<>();
```
Can we please add a log message here?
This allows for better debugging in case we have some problems with the migration

muyangye replied:
Just added a log message. By the way even if this exception occurs, rest of the files' migration will not be affected since it will just be skipped.

bossenti reviewed (commented):
@muyangye Thank you very much for your dedicated work!
I have only two last minor remarks

bossenti commented on ui/cypress.config.ts:
```diff
@@ -44,7 +44,7 @@ export default defineConfig({
         },
         specPattern: 'cypress/tests/**/*.{js,jsx,ts,tsx}',
-        baseUrl: 'http://localhost:80',
+        baseUrl: 'http://localhost:8082',
```
I assume this was committed by accident, right?

muyangye replied:
Ah yes, I accidentally did "git commit -am" instead of only committing one file. I have reverted all the irrelevant changes. Thank you for catching it!

bossenti commented:
Thank you very much for your contribution here @muyangye 🙏🏼
